File: app/src/main/java/team58/cs2340/donationtracker/models/Category.java
package team58.cs2340.donationtracker.models;

public enum Category {
    APPLIANCES("Appliances"),
    BABY("Baby"),
    BAGSANDACCESSORIES("Bags and Accessories"),
    BOOKSANDMUSIC("Books and Music"),
    CLOTHING("Clothing"),
    ELECTRONICS("Electronics"),
    FOOD("Food"),
    FURNITURE("Furniture"),
    MOVIESANDGAMES("Movies and Games"),
    SPORTSANDOUTDOORS("Sports and Outdoors"),
    TOYS("Toys");



    private final String categoryName;

    Category(String categoryName) {
        this.categoryName = categoryName;
    }

    @Override
    public String toString() {
        return this.categoryName;
    }

    /**
     * Function to convert string to enum
     * @param text string to be converted
     * @return enum of string
     */
    public static Category fromString(String text) {
        for (Category c : Category.values()) {
            if (c.categoryName.equalsIgnoreCase(text)) {
                return c;
            }
        }
        return null;
    }
}

File: app/src/main/java/team58/cs2340/donationtracker/models/Category.kt
package team58.cs2340.donationtracker.models

/**
 * Category enum
 */
enum class Category private constructor(private val categoryName: String) {
    APPLIANCES("Appliances"),
    BABY("Baby"),
    BAGSANDACCESSORIES("Bags and Accessories"),
    BOOKSANDMUSIC("Books and Music"),
    CLOTHING("Clothing"),
    ELECTRONICS("Electronics"),
    FOOD("Food"),
    FURNITURE("Furniture"),
    MOVIESANDGAMES("Movies and Games"),
    SPORTSANDOUTDOORS("Sports and Outdoors"),
    TOYS("Toys");

    override fun toString(): String {
        return this.categoryName
    }

    companion object {

        /**
         * Function to convert string to enum
         * @param text string to be converted
         * @return enum of string
         */
        fun fromString(text: String): Category? {
            for (c in Category.values()) {
                if (c.categoryName.equals(text, ignoreCase = true)) {
                    return c
                }
            }
            return null
        }
    }
}
